pkg/aws/ec2/pricing_map.go

```diff
@@ -127,40 +150,71 @@ func (spm *StructuredPricingMap) GeneratePricingMap(ondemandPrices []string, spo
 	return nil
 }
 
-// AddToPricingMap adds a price to the pricing map. The price is weighted based upon the instance type's CPU and RAM.
-func (spm *StructuredPricingMap) AddToPricingMap(price float64, attribute Attributes) error {
-	spm.m.Lock()
-	defer spm.m.Unlock()
-	if spm.Regions[attribute.Region] == nil {
-		spm.Regions[attribute.Region] = &FamilyPricing{}
-		spm.Regions[attribute.Region].Family = make(map[string]*Prices)
+// GenerateStoragePricingMap receives a json with all the prices of the available storage options
+// It iterates over the storage classes and parses the price for each one.
+func (spm *StoragePricingMap) GenerateStoragePricingMap(storagePrices []string) error {
```

Comment:
At some point (probably at the start of this function), this should lock the Mutex on `StoragePricingMap`, unless it's already locked somewhere upstream and I missed that.